src/docs.ts

```diff
@@ -54,4 +55,42 @@ export class Docs {
 
     return parent;
   }
+
+  private static getPatientPlaceIdentifiers(parentDoc: Doc) {
+    if (parentDoc.type === DocType.person || parentDoc.contact_type === DocType.person) {
+      return {
+        patient_id: parentDoc.patient_id,
+        patient_uuid: parentDoc._id,
+      };
+    }
+    return {
+      place_id: parentDoc.place_id,
+      place_uuid: parentDoc._id,
+    };
+  }
+
+  private static populateDataRecordParentData(doc, parentDoc: Doc) {
+    return {
+      contact: this.createParentRelation(parentDoc),
+      fields: {
+        ...this.getPatientPlaceIdentifiers(parentDoc),
+        ...doc.fields,
+      }
+    };
+  }
+
+  private static getParentAssociationData(doc, parentDoc: Doc) {
+    if (doc.type === DocType.dataRecord) {
+      const dataRecordParent = doc.contact || parentDoc;
+      if (!dataRecordParent) {
+        return {};
+      }
+      return this.populateDataRecordParentData(doc, dataRecordParent);
+    }
```

Comment:
```suggestion
    if (doc.type === DocType.dataRecord) {
      const dataRecordParent = doc.contact || parentDoc;
      return dataRecordParent ? this.populateDataRecordParentData(doc, dataRecordParent) : {};
    }
```
Minor, in case you want to avoid nested `if` and still easy to read. Otherwise it's okay